src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -0,0 +1,145 @@
+
+bool InsertSpecificIterations::run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, lowered::LinearIR::constExprIt end) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InsertSpecificIterations")
+    const auto& loop_manager = linear_ir.get_loop_manager();
+
+    bool modified = false;
+    for (auto expr_it = begin; expr_it != end; ++expr_it) {
+        const auto& expr = *expr_it;
+        const auto node = expr->get_node();
+        const auto loop_end = ov::as_type_ptr<op::LoopEnd>(node);
+        if (!loop_end)
+            continue;
+
+        const auto& loop_info = loop_manager->get_loop_info(loop_end->get_id());
+        const auto work_amount = loop_info->get_work_amount();
+        const auto increment = loop_info->get_increment();
+        const auto& handlers = loop_info->get_handlers();
+
+        const auto main_loop_begin_it = linear_ir.find(linear_ir.get_expr_by_node(loop_end->get_loop_begin()));
+        const auto main_loop_end_it = linear_ir.find_after(main_loop_begin_it, linear_ir.get_expr_by_node(loop_end));
+        // Note: handlers must be run on the range started with the first operation in the loop body.
+        const auto main_first_body_op_it = std::next(main_loop_begin_it);
+
+        auto update_loop_params = [&loop_manager](const std::shared_ptr<op::LoopEnd>& loop_end_copy,
+                                                  size_t new_work_amount,
+                                                  size_t new_increment,
+                                                  bool zero_finalization_offsets) {
+            loop_end_copy->set_work_amount(new_work_amount);
+            loop_end_copy->set_increment(new_increment);
+
+            const auto& loop_info_copy = loop_manager->get_loop_info(loop_end_copy->get_id());
+            loop_info_copy->set_work_amount(new_work_amount);
+            loop_info_copy->set_increment(new_increment);
+
+            if (zero_finalization_offsets)
+                loop_end_copy->set_finalization_offsets(std::vector<int64_t>(loop_end_copy->get_finalization_offsets().size(), 0));
+        };
+
+        auto copy_and_run_specific_handlers = [&](const PassPipeline& handlers) {
+            const auto& cloned_body = copy_loop(linear_ir, loop_end->get_id());
+            lowered::LinearIR::constExprIt start = linear_ir.insert(main_loop_begin_it, cloned_body.begin(), cloned_body.end());
+            const auto cloned_loop_end = *std::prev(cloned_body.end());
+            auto end = linear_ir.find_after(start, cloned_loop_end);
+            // Note: handlers must be run on the range started with the first operation in the loop body.
+            handlers.run(linear_ir, std::next(start), end);
+            return ov::as_type_ptr<op::LoopEnd>(cloned_loop_end->get_node());
+        };
+
+        const bool specific_first_iteration = !handlers.get_first_iter_handelrs().empty();
+        if (work_amount == increment) {
```

Comment:
Does it mean that the passes should control tail transformation themself now? 
For example, if the transformation has the specific changes in cases when `WA < vector_size`, it should be covered by this transformation?

Before these transformations (`tail_transformations`) were called separately. But seems like we should control it in passes now.

For example, in `InsertLoadStore` we set count `min(dim, vector_size)` instead of `1 or vector_size`. In `SoftmaxDecomposition` we insert `Fill` ops with offset `min(dim, vector_size)`.

I'm not sure that it's safe (to always remember in new pass implementing). Or I just used to rely on `tail_transformation`.

If it's so and it's final decision, let's to update the doc related to passes. Otherwise new developers can miss this nuance.
What do you think? Also @IvanNovoselov probably you have some thoughts regarding that as well? 😃

Reply:
We anyway have to control tail transformation themself in lowered passes since we need to set parameters for the corresponding specific iteration handlers (most of them requires `tail_size` as a parameter). Basing on that, we also decided to avoid loop creation with `increment > work_amount` (I already vaguely remember, but such case sometimes required special handling: maybe @IvanNovoselov can share some examples which was met during TPP related activities)
I agree that the discussed changes must be reflected in the docs. However, as I said below, I would prefer to avoid potential merge conflicts with #22258, so I would like to update docs in a separate PR. If #22258 is merged before this PR, I will update it.